StringCatalogEnum/Sources/StringCatalogEnum/File.swift

```diff
@@ -0,0 +1,156 @@
+        let parser = JSONParser()
+        let stringsDictionary = try parser.parseJSON(jsonData)
+        let codeGenerator = EnumCodeGenerator(enumName: enumName, enumTypealias: enumTypealias)
+        let enumCode = codeGenerator.generateEnumCode(from: stringsDictionary)
+        try fileManager.writeOutput(to: outputFilename, content: enumCode)
+    }
+}
+
+class FileManager {
+    func readDataFromFile(_ path: String) throws -> Data {
+        let fileURL = URL(fileURLWithPath: path)
+        do {
+            let data = try Data(contentsOf: fileURL)
+            return data
+        } catch {
+            throw error
+        }
+    }
+
+    func writeOutput(to filename: String, content: String) throws {
+        let fileURL = URL(fileURLWithPath: filename)
+        do {
+            try content.write(to: fileURL, atomically: true, encoding: .utf8)
+        } catch {
+            throw error
+        }
+    }
+}
+
+class JSONParser {
+    func parseJSON(_ data: Data) throws -> [String: Any] {
+        do {
+            let jsonObject = try JSONSerialization.jsonObject(with: data, options: [])
+            guard let dictionary = jsonObject as? [String: Any] else {
+                throw StringCatalogEnum.Error.unexpectedJSON(message: "Data is not a JSON dictionary")
+            }
+            return dictionary
+        } catch {
+            throw error
+        }
+    }
+}
+
+
+class EnumCodeGenerator {
+    var enumName: String
+    var enumTypealias: String
+
+    init(enumName: String, enumTypealias: String) {
+        self.enumName = enumName
+        self.enumTypealias = enumTypealias
+    }
+
+    func generateEnumCode(from dictionary: [String: Any]) -> String {
+        var enumCode = """
+            // This file is generated by XcodeStringEnum. Please do *NOT* update it manually.
+            // As a common practice, swiftLint is disabled for generated files.
+            // swiftlint:disable all
+
+            import SwiftUI
+
+            /// Makes it a bit easier to type.
+            typealias \(enumTypealias) = \(enumName)
+
+            /// Generated by StringCatalogEnum, this enum contains all existing String Catalog keys.
+            enum \(enumName): String, CaseIterable {
+            """
+
+        var cases = [String]()
+        for (key, _) in dictionary {
+            guard let name = convertToVariableName(key: key) else {
+                continue
+            }
+            if StringCatalogEnum.Keyword.allCases.map({ $0.rawValue }).contains(name) {
+                cases.append("    case `\(name)`\n")
+            } else {
+                cases.append("    case \(name)\n")
+            }
+        }
+
+        cases.sort()
+        cases.forEach { string in
+            enumCode += string
+        }
+
+        enumCode += """
+            }
+            // swiftlint:enable all
+            """
+
+        return enumCode
+    }
+
+    private func convertToVariableName(key: String) -> String? {
```

Comment:
Actually I was expecting you to test this function... It's interesting that you chose the other code to test. No problem with that, but case conversion is definitely the component we want to unit test the most.